lib/game.rb

```diff
@@ -0,0 +1,92 @@
+class Game
+  @@v_conditions = [[1,2,3], [4,5,6], [7,8,9], [1,4,7], [2,5,8], [3,6,9], [1,5,9], [3,5,7]]
+  attr_reader :board
+
+  def initialize(p1, p2)
+    @p1 = p1
+    @p2 = p2
+    @board = Board.new
+    @active_player = rand()*10 < 5 ? @p2 : @p1
+  end
+
+  def cycle
+    loop do
+      change_active_player
+      @active_player.turn(self)
+      show(@board)
+      if victory?(@active_player)
+        @active_player.score_up
+        win_msg(@active_player)
+        break
+      end
+      if tie?
+        tie_msg
+        break
+      end
+    end
+  end
+
+  def restart?
```

Comment:
Things about this method:

1. It combines UI logic with game related logic
2. It's probably not a responsibility anymore of the game object. What you can do is your tic-tac-toe.rb file should be creating a new instance of board, player1, player2 and game everytime the user wants to play again.